File: src/main/java/com/server/ordering/domain/dto/response/BasketFood.java
package com.server.ordering.domain.dto.response;

import com.server.ordering.domain.Basket;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class BasketFood {

    private Long basketId;
    private Long foodId;
    private String foodName;
    private String imageUrl;
    private Boolean soldOut;
    private int price;
    private int count;

    public BasketFood(Basket basket) {
        this.basketId = basket.getId();
        this.foodId = basket.getFood().getId();
        this.foodName = basket.getFood().getFoodName();
        this.imageUrl = basket.getFood().getImageUrl();
        this.soldOut = basket.getFood().isSoldOut();
        this.price = basket.getFood().getPrice();
        this.count = basket.getCount();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/response/BasketFood.kt
package com.byeoru.ordering_server.domain.dto.response

import com.byeoru.ordering_server.domain.Basket

class BasketFood(basket: Basket) {

    val basketId: Long?
    val foodId: Long?
    val foodName: String
    val imageUrl: String?
    val soldOut: Boolean
    val price: Int
    val count: Int

    init {
        this.basketId = basket.id
        this.foodId = basket.food.id
        this.foodName = basket.food.foodName
        this.imageUrl = basket.food.imageUrl
        this.soldOut = basket.food.isSoldOut
        this.price = basket.food.price
        this.count = basket.count
    }
}